website/project/model.py

```diff
@@ -1689,10 +1689,20 @@ def save(self, *args, **kwargs):
         if 'node_license' in saved_fields:
             children = [c for c in self.get_descendants_recursive(
                 include=lambda n: n.node_license is None
-            )]
+            ) if c.is_public and not c.is_deleted]
             # this returns generator, that would get unspooled anyways
             if children:
-                Node.bulk_update_search(children)
+                count = len(children) / 100
+                if count == 0:
+                    Node.bulk_update_search(children)
+                else:
+                    start = 0
+                    end = 100
+                    for i in range(1, count + 1):
+                        Node.bulk_update_search(children[start:end])
+                        start = end
+                        end = 100 * (i + 1)
+                    Node.bulk_update_search(children[(100 * count):])
```

Comment:
This looks like it might hit a problem on an edge case, so we'd either want to break it out into a function and unit test that function, or we can possibly simplify with something like:

```py
while len(children):
    batch = children[:99]
    children = children[99:]
    Node.bulk_update_search(batch)
```